packages/react-router-dev/vite/config.ts

```diff
@@ -412,26 +412,40 @@ export async function resolveReactRouterConfig({
   };
 
   let routeConfigFile = findEntry(appDirectory, "routes");
-  if (!routeConfigFile) {
-    let routesConfigDisplayPath = path.relative(
-      rootDirectory,
-      path.join(appDirectory, "routes.ts")
-    );
-    logger.error(
-      colors.red(
-        `Could not find a routes config file at "${routesConfigDisplayPath}"`
-      )
-    );
-    process.exit(1);
-  }
+
+  class FriendlyError extends Error {}
 
   try {
+    if (!routeConfigFile) {
+      let routesConfigDisplayPath = path.relative(
+        rootDirectory,
+        path.join(appDirectory, "routes.ts")
+      );
+      throw new FriendlyError(
+        `Could not find a routes config file at "${routesConfigDisplayPath}"`
+      );
+    }
+
     setAppDirectory(appDirectory);
-    let routesConfig: RoutesConfig = (
+    let routesConfigExport: RoutesConfig = (
       await viteNodeRunner.executeFile(path.join(appDirectory, routeConfigFile))
-    ).default;
+    ).routes;
+
+    let routesConfig = await routesConfigExport;
+
+    if (!routesConfig) {
+      throw new FriendlyError(
+        `No "routes" export defined in "${routeConfigFile}"`
```

Comment:
Changing the export made me realise that we could improve the error handling around cases like this, so I've improved the error handling a little.